### PR TITLE
Pyed is merge with Tyr and become multitenant

### PR DESCRIPTION
**Pull request by kinnou02**
Now only one Tyr is needed for all the ED.
- the task are run with celery
- the databases is now construct with Alembic 
- The models and the protobuf are in the python package navitiacommon

TODO:
- Deployement
- API for display jobs
- add an authentification to Tyr for administrative task?
- new realtime message don't trigger nav2rt

DON'T MERGE BEFORE THE CURRENT RELEASE IS GOOD!!

### CODE REVIEW

File: source/tyr/tyr/binarisation.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        job.state = 'failed'
+        models.db.session.commit()
+        raise
+    finally:
+        lock.release()
+
+
+@celery.task()
+def osm2ed(instance_config, osm_filename, job_id):
+    """ Move osm file to backup directory, launch osm2ed """
```
Comment (l-vincent-l) [inline source/tyr/tyr/binarisation.py]: En fait le move est fait en amont

File: source/tyr/tyr/binarisation.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        job.state = 'failed'
+        models.db.session.commit()
+        raise
+    finally:
+        lock.release()
+
+
+@celery.task()
+def gtfs2ed(instance_config, gtfs_filename,  job_id):
+    """ Unzip gtfs file, remove the file, launch gtfs2ed """
```
Comment (l-vincent-l) [inline source/tyr/tyr/binarisation.py]: il faut enlever le remove, en fait il est move avant

File: source/tyr/tyr/binarisation.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+    connection_string += ' user=' + instance_config.pg_username
+    connection_string += ' dbname=' + instance_config.pg_dbname
+    connection_string += ' password=' + instance_config.pg_password
+    return connection_string
+
+
+#TODO bind task
+@celery.task()
+def fusio2ed(instance_config, filename, job_id):
+    """ Unzip fusio file, remove the file, launch fusio2ed """
```
Comment (l-vincent-l) [inline source/tyr/tyr/binarisation.py]: il faut enlever le remove, en fait il est move avant

File: source/tyr/tyr/helper.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+from configobj import ConfigObj, flatten_errors
+from validate import Validator
+from flask import current_app
+from collections import namedtuple
+
+
+def configure_logger(app):
+    """
+    configuration du logging
+    """
```
Comment (l-vincent-l) [inline source/tyr/tyr/helper.py:13]: English please!

File: source/tyr/tyr/helper.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    logging.getLogger('sqlalchemy.engine').setLevel(
+            app.config["LOG_LEVEL_SQLALCHEMY"])
+    logging.getLogger('sqlalchemy.pool').setLevel(
+            app.config["LOG_LEVEL_SQLALCHEMY"])
+    logging.getLogger('sqlalchemy.dialects.postgresql')\
+            .setLevel(app.config["LOG_LEVEL_SQLALCHEMY"])
+
+    logging.getLogger('sqlalchemy.engine').addHandler(handler)
+    logging.getLogger('sqlalchemy.pool').addHandler(handler)
+    logging.getLogger('sqlalchemy.dialects.postgresql').addHandler(handler)
```
Comment (l-vincent-l) [inline source/tyr/tyr/helper.py:31]: So all the logs are going in one file ?

File: source/tyr/tyr/resources.py [hunk 6]
```diff
@@ -96,15 +97,15 @@ def put(self, user_id):
...
         try:
             user.email = args['email']
             db.session.commit()
             return marshal(user, user_fields_full)
-        except sqlalchemy.exc.IntegrityError as e:
+        except sqlalchemy.exc.IntegrityError:
             return ({'error': 'duplicate user'}, 409)  # Conflict
-        except Exception as e:
+        except Exception:
             logging.exception("fail")
```
Comment (l-vincent-l) [inline source/tyr/tyr/resources.py:109]: ça va être pratique à debugger ça tient

File: source/tyr/tyr/resources.py [hunk 7]
```diff
@@ -133,14 +136,13 @@ def get(self, user_id, key_id=None):
     def post(self, user_id):
         parser = reqparse.RequestParser()
         parser.add_argument('valid_until', type=types.date, required=False,
-                            case_sensitive=False,
-                            help='date de fin de validité de la clé')
+                case_sensitive=False, help='date de fin de validité de la clé')
```
Comment (l-vincent-l) [inline source/tyr/tyr/resources.py]: English please!

File: source/tyr/tyr/resources.py [hunk 8]
```diff
@@ -163,8 +165,7 @@ def delete(self, user_id, key_id):
     def put(self, user_id, key_id):
         parser = reqparse.RequestParser()
         parser.add_argument('valid_until', type=types.date, required=True,
-                            case_sensitive=False,
-                            help='date de fin de validité de la clé')
+                case_sensitive=False, help='date de fin de validité de la clé')
```
Comment (l-vincent-l) [inline source/tyr/tyr/resources.py]: English please!

File: source/tyr/tyr/resources.py (hunk 6 above)
Comment (kinnou02) [inline source/tyr/tyr/resources.py:109]: logging.exeption récupére automatiquement l'exception courante et la log avec la stacktrace associé

File: source/tyr/tyr/helper.py (hunk 5 above)
Comment (kinnou02) [inline source/tyr/tyr/helper.py:31]: Yes it's easier if we don't have to look at five files for know what happens.
But the logs of binary like osm2ed or ed2nav should be put in another file, probably one per instances

**Pull request merged**